src/_macosx.m

```diff
@@ -629,6 +630,39 @@ static CGFloat _get_device_scale(CGContextRef cr)
     return (PyObject*)self;
 }
 
+int set_icon(PyObject* icon_path) {
+    if (!icon_path) {
+        return 0;
+    }
+    const char* icon_path_ptr = PyUnicode_AsUTF8(icon_path);
+    if (!icon_path_ptr) {
+        Py_DECREF(icon_path);
+        return -1;
+    }
+    NSString* ns_icon_path = [NSString stringWithUTF8String: icon_path_ptr];
+    if (!ns_icon_path) {
+        Py_DECREF(icon_path);
+        PyErr_SetString(PyExc_RuntimeError, "Could not convert to NSString*");
+        return -1;
+    }
+    NSImage* image = [[NSImage alloc] initByReferencingFile: ns_icon_path];
```

Comment:
Can you add `[... autorelease]` here and let the program do the releasing for you?

Reply:
I'd have to add an @autorelease {} scope block, right?

Reply:
I am not an expert in objc either... But, I _think_ your new update is the way to do it and the scope will be accounted for properly. The autorelease pools are optional if you want finer grained control I think.